runtime/doc/lua.txt

```diff
@@ -2797,7 +2797,7 @@ vim.keymap.del({modes}, {lhs}, {opts})                      *vim.keymap.del()*
     See also: ~
       • |vim.keymap.set()|
 
-vim.keymap.set({mode}, {lhs}, {rhs}, {opts})                *vim.keymap.set()*
+vim.keymap.set({mode}, {lhs}, {opts}, {rhs})                *vim.keymap.set()*
```

Comment:
After some more thought, let's make it like this:

```suggestion
vim.keymap.set({mode}, {opts}, {lhs}, {rhs})                *vim.keymap.set()*
```

that should resolve the objections?

Reply:
I don't think it does resolve the objections. I think people are happy with the current signature and `opts` at the end works the best for most people. Putting it anywhere else is at best a sideways change, and at worst a regression for a lot of mappings.

If you _really_ want to put `rhs` at the end then the only option I see is:

```
vim.keymap.set({mode}, [{opts},] {lhs}, {rhs})
```

So `opts` is optional.

But I'm sure this will bring other objections.

Reply:
> If you _really_ want to put `rhs` at the end then the only option I see is:

that's exactly the idea. overloads are normal and cheap, if anyone objects to this they are a bot.